core/src/main/clojure/xtdb/pgwire.clj

```diff
@@ -598,7 +598,9 @@
 (def ^:private io-uint32
   "An unsigned 32bit integer"
   {:read #(.readInt ^DataInputStream %)
-   :write #(.writeInt ^DataOutputStream %1 (int %2))})
+   :write (fn [^DataOutputStream out v]
+              (when v
+                (.writeInt out (int v))))})
```

Comment:
I think this function should fail rather than not write anything in case of nil. Why is this change needed anyway?